From c715edfe85dc5683225b762552d2a3b1beba33aa Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Fri, 25 Sep 2020 10:47:45 +0800
Subject: [PATCH] ARM: dts: rockchip: get sensor data without vicap for
 rv1126-bat-v10

Plan A: MIPI/LVDS-->VICAP-->ISP-->ISPP
Plan B: MIPI/LVDS-->ISP-->ISPP

The performance of Plan B is enough for 1080P sensor. Moreover, it saves
about 12+mw than Plan A.

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
Change-Id: I5abbc76e831625b822c6b15b90f0b2c8e530483d
---
 arch/arm/boot/dts/rv1126-bat-ipc-v10.dts | 69 +-----------------------
 1 file changed, 2 insertions(+), 67 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
index fc65695e5e4b..806c9fd38807 100644
--- a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
+++ b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
@@ -172,7 +172,7 @@
 
 			csidphy0_out: endpoint@0 {
 				reg = <0>;
-				remote-endpoint = <&mipi_csi2_input>;
+				remote-endpoint = <&isp_in>;
 			};
 		};
 	};
@@ -292,39 +292,6 @@
 	npu-supply = <&vdd_npu_vepu>;
 };
 
-&mipi_csi2 {
-	status = "okay";
-
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		port@0 {
-			reg = <0>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			mipi_csi2_input: endpoint@1 {
-				reg = <1>;
-				remote-endpoint = <&csidphy0_out>;
-				data-lanes = <1 2>;
-			};
-		};
-
-		port@1 {
-			reg = <1>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			mipi_csi2_output: endpoint@0 {
-				reg = <0>;
-				remote-endpoint = <&cif_mipi_in>;
-				data-lanes = <1 2>;
-			};
-		};
-	};
-};
-
 &pinctrl {
 	es8311 {
 		spk_ctl: spk-ctl {
@@ -364,34 +331,6 @@
 	vccio7-supply = <&vcc_3v3>;
 };
 
-&rkcif {
-	status = "okay";
-};
-
-&rkcif_mipi_lvds {
-	status = "okay";
-
-	port {
-		/* MIPI CSI-2 endpoint */
-		cif_mipi_in: endpoint {
-			remote-endpoint = <&mipi_csi2_output>;
-			data-lanes = <1 2>;
-		};
-	};
-};
-
-&rkcif_mipi_lvds_sditf {
-	status = "okay";
-
-	port {
-		/* MIPI CSI-2 endpoint */
-		mipi_lvds_sditf: endpoint {
-			remote-endpoint = <&isp_in>;
-			data-lanes = <1 2>;
-		};
-	};
-};
-
 &rkisp_thunderboot {
 	reg = <0x10000000 (72 * 0x00100000)>;
 };
@@ -407,7 +346,7 @@
 
 			isp_in: endpoint@0 {
 				reg = <0>;
-				remote-endpoint = <&mipi_lvds_sditf>;
+				remote-endpoint = <&csidphy0_out>;
 			};
 		};
 	};
@@ -417,10 +356,6 @@
 	status = "okay";
 };
 
-&rkispp_vir0 {
-	status = "okay";
-};
-
 &rkispp_mmu {
 	status = "okay";
 };
-- 
2.35.3

